### DISCUSSION TITLE
visualize voxel

### DISCUSSION THREAD

**Discussion by JeffreyYH** [AcademySoftwareFoundation/openvdb#1676] [General]
How to visualize the voxel grids or .vdb file generated by openVDB? When I #include <openvdb/viewer/Viewer.h>, it shows no such file or directory

**Comment by DragonLiu1** [comment]
Houdini

**Comment by Idclip** [comment]
Hello @JeffreyYH - the Viewer files are not installed as part of the OpenVDB installation (this is a bit weird I know), so I'm not really sure if they were ever meant to be fully downstreamed as a finished public API. The files in `openvdb/viewer` are only used to build the `vdb_view` (openvdb_view.cc) binary. You can use this binary or various DCCs (such as Houdini) to visualize a VDB. Should you wish to build your own application that visualizes VDBs, you'll have to copy the entire viewer directory and incorporate it into your project.

We could add an option to build this support into `libopenvdb.so` or build a separate `libopenvdb_view.so` which incorporates this functionality. Would require more discussion.
